crates/krun-guest/src/pulse.rs

```diff
@@ -0,0 +1,24 @@
+use std::{fs, path::PathBuf, process::Command};
+
+use anyhow::{Context, Result};
+use utils::env::find_in_path;
+
+pub fn setup_pulse_proxy(run_path: PathBuf) -> Result<()> {
+    if let Some(socat_path) = find_in_path("socat")? {
+        let mut pulse_path = run_path.join("pulse");
+        fs::create_dir(&pulse_path)
+            .context("Failed to create `pulse` directory in `XDG_RUNTIME_DIR`")?;
+        pulse_path = pulse_path.join("native");
```

Comment:
Oh, I mistakenly thought `join` takes a `&mut self`.

This can be changed to:
```suggestion
        let pulse_path = pulse_path.join("native");
```

And we can get rid of the `mut` above.